test/functional/tests/ci/test_recovery.py

```diff
@@ -0,0 +1,227 @@
+        core_devs = core_disk.partitions
+
+    with TestRun.step("Start caches and add cores."):
+        caches, cores = [], []
+        for (cache_dev, core_dev) in zip(cache_devs, core_devs):
+            cache = casadm.start_cache(cache_dev, cache_mode=CacheMode.WB)
+            core = cache.add_core(core_dev)
+            caches.append(cache)
+            cores.append(core)
+
+    with TestRun.step("Create filesystem on core devices."):
+        filesystems = [Filesystem.xfs, Filesystem.ext3, Filesystem.ext4]
+        for (core, filesystem) in zip(cores, filesystems):
+            core.create_filesystem(filesystem)
+
+    with TestRun.step("Mount cache devices."):
+        core_mount_points = []
+        for (cache, core) in zip(caches, cores):
+            core_mnt_point = f"{mount_point}-{cache.cache_id}-{core.core_id}"
+            core_mount_points.append(core_mnt_point)
+            core.mount(core_mnt_point)
+
+            with TestRun.step("Run IO"):
+                dd = (
+                    Dd()
+                    .input("/dev/urandom")
+                    .output(f"{core_mnt_point}/test")
+                    .count(1)
+                    .block_size(Size(50, Unit.MegaByte))
+                )
+                dd.run()
+                sync()
+
+    with TestRun.step("Calculate cache md5sums before unplug."):
+        core_mnt_md5s_before = [
+            File(f"{core_mnt_point}/test").md5sum() for core_mnt_point in core_mount_points
+        ]
+
+    with TestRun.step("Umount core devices."):
+        for core in cores:
+            core.unmount()
+
+    with TestRun.step("Dirty stop"):
+        dirty_stop(cache_disk, caches)
+
+    with TestRun.step("Start caches with load metadata and later stop them."):
+        for cache_dev in cache_devs:
+            cache = casadm.load_cache(cache_dev)
+            cache.stop()
+
+    with TestRun.step("Mount core devices."):
+        core_mnt_points = []
+        for core, cache in zip(cores, caches):
+            core_mnt_point = f"{mount_point}-{cache.cache_id}-{core.core_id}"
+            core_mnt_points.append(core_mnt_point)
+            mount(core.core_device, core_mnt_point)
+            if not check_if_file_exists(f"{core_mnt_point}/test"):
+                TestRun.LOGGER.error(f"Mounting core device {core_mnt_point} failed.")
+
+    with TestRun.step("Calculate cache md5sums after recovery."):
+        core_mnt_md5s_after = [
+            File(f"{core_mnt_point}/test").md5sum() for core_mnt_point in core_mnt_points
+        ]
+
+    with TestRun.step("Compare md5 sums for cores and core devices"):
+        if core_mnt_md5s_before != core_mnt_md5s_after:
+            TestRun.fail(f"MD5 sums of core before and after does not match.")
+
+    with TestRun.step("Umount core devices."):
+        for core_dev in core_devs:
+            unmount(core_dev)
+
+    with TestRun.step("Dirty stop"):
+        dirty_stop(cache_disk, caches)
```

Comment:
was it there in the original test? seems to be unnecessary

Reply:
Yes it was but prepare and clean up form test framework may be enough.